crypto/merkle.go

```diff
@@ -124,10 +124,12 @@ func MerkleProof(b []byte, proofIndex uint64) (base []byte, hashSet []Hash) {
 
 	// Get the proof and convert it to a base + hash set.
 	_, proof, _, _ := t.Prove()
-	base = proof[0]
-	hashSet = make([]Hash, len(proof)-1)
-	for i, p := range proof[1:] {
-		copy(hashSet[i][:], p)
+	if len(proof) > 0 {
```

Comment:
it's only a problem when `len(b) == 0` as far as I can see. But it was causing a panic during testing. We could probably throw that check at the top if you think that's cleaner.